google/cloud/internal/log_impl.cc

```diff
@@ -23,6 +23,7 @@ namespace internal {
 void StdClogBackend::Process(LogRecord const& lr) {
   std::lock_guard<std::mutex> lk(mu_);
   if (lr.severity >= min_severity_) {
+    std::cout << std::flush;
```

Comment:
> Sorry for continuing to nit pick (and to be slow to do so!), but I imagined that you would only want to do this as often as you flushed `std::clog`. That is, within the `lr.severity` conditional, and also in `Flush()` itself.

I am not sure that is the case.  I want to preserve some notion of "happened-before".  The messages to `std::cout` sent before the message to `std::clog` should be displayed before, and thus the `std::cout << std::flush` should precede the `std::clog << lr << "\n"`.

> 
> I also thought your previous idea of also flushing `std::cerr` was a good one. It might be block buffered.

Done.

Reply:
> I want to preserve some notion of "happened-before".

I think that notion is long-since lost if there is any buffering going on, but OK.